api/api/models/moderation.py

```diff
@@ -0,0 +1,35 @@
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+
+class UserPreferences(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    preferences = models.JSONField(default=dict)
+
+    def __str__(self):
+        return f"{self.user.username}'s preferences"
+
+    @property
+    def moderator(self):
+        return self.preferences.get("moderator", {})
```

Comment:
I realised yesterday thinking about this PR that a small bug could come from this particular implementation of the `moderator` getter, caused by sloppy object reference handling (my bad!):

```py
assert "moderator" not in prefs.preferences
prefs.moderator["property"] = "whatever"
prefs.save()
```

`prefs.moderator` is the default case from `get`, rather than the actual object on `self.preferences`, and we haven't "bound" that default object to `self.preferences`. Therefore, this won't actually change the `preferences` object at all, because the moderator object won't be saved into the `preferences` object. The only reason there isn't a bug in the code we have already is because the `blur_images` setter uses `|=` which will call the setter for moderator. But if you assigned directly to `self.moderator["blur_images"]`, it wouldn't work.

We can fix this by:

```suggestion
        if "moderator" not in self.preferences:
            self.preferences["moderator"] = {}
        
        return self.preferences["moderator"]
```

That way, the return value of the moderator getter is a reference to the object on `self.preferences` and never a "free" object.